koji/collection-mash-split.py

```diff
@@ -373,6 +373,22 @@ def run_mashes(collection, git_tag, mashes):
                        mash_config.compses, git_tag)
 
 
+def get_foreman_version(katello_version):
+    branch_map = {
+        'nightly': 'develop',
+        '4.1': '2.5',
+        '4.0': '2.4',
+        '3.18': '2.3',
+    }
+
+    try:
+        return branch_map[katello_version]
+    except KeyError:
+        major, minor = map(int, katello_version.split('.', 1))
+        # Katello (4 + x).(2 + y) -> Foreman (3 + x).(0 + y)
+        return "{}.{}".format(major - 1, minor - 2)
```

Comment:
This assumes that Katello will keep its 4.y and Foreman its 3.y for a while, which I don't feel like is something we should rely on.

@tbrisker @jlsherrill @ehelms what do y'all think?

Reply:
The counter argument is that now we need to add every version. That means every branching it needs an update. If we do have a major version bump in either project or if there's more than 1 Katello release per Foreman release, we can then deal with it.

Reply:
Yeah, this is my main concern:
Right now, the procedure is easy "go and update the mapping"
With this change, it would become "go and see if the mapping needs updating" and if that wasn't needed for a few releases, people will skip this step in their minds.

That all said, I am not doing releases, so :woman_shrugging:

Reply:
We have [a line in Foreman's branching](https://github.com/theforeman/tool_belt/blob/aec8772665dab55ce4e056f8f5c5c66936e64bd8/procedures/foreman/branch.md.erb#L80) and then there's also [instructions in Katello's branching](https://github.com/theforeman/tool_belt/blob/aec8772665dab55ce4e056f8f5c5c66936e64bd8/procedures/katello/branch.md.erb#L99) but that's outdated and refers to a file that hasn't existed since [March 2020](https://github.com/theforeman/foreman-infra/commit/c64ede30cd542b4adbf3c1da1765efcefbbd8829).

That either means people ignore the procedures or don't update them when they're incorrect. Given I wonder what's best.

Reply:
Ole Murphy says that as soon as we make this change one of the projects will break the assumption :grin: 

We have multiple places in the project that need the mapping of Foreman version to Katello version -- is there some central place we could store it and then access it so we update it in one place?